pkg/types/validation/machinepools.go

```diff
@@ -38,7 +38,7 @@ func ValidateMachinePool(p *types.MachinePool, fldPath *field.Path, platform str
 		allErrs = append(allErrs, field.NotSupported(fldPath.Child("name"), p.Name, validMachinePoolNameValues))
 	}
 	if p.Replicas != nil {
-		if *p.Replicas <= 0 {
+		if *p.Replicas < 0 {
 			allErrs = append(allErrs, field.Invalid(fldPath.Child("replicas"), p.Replicas, "number of replicas must be positive"))
```

Comment:
Change the text in the error generated to reflect that the number of replicas must not be negative.